### DISCUSSION TITLE
How are Eloquent Filters supposed to be used?

### DISCUSSION THREAD

**Discussion by caposaric** [orchidsoftware/platform#2590] [Q&A]
I'm trying to implement a custom filter for a relationship field that is displayed on the table, but just can't get it working. Can this be done without resorting to Selections?

The intended functionality: filter the tests from a specific user (by email)

UserEmailFilter.php
```php
    public function parameters(): ?array
    {
        return ['user_email'];
    }

    public function run(Builder $builder): Builder
    {
        dd($this->request->get('user_email'));
        return $builder->whereHas('user', function ($query) {
            $query->where('email', 'LIKE', '%' . $this->request->get('user_email') . '%');
        });
    }

    public function display(): iterable
    {
        return [
            Input::make('user_email')
                ->type('text')
                ->value($this->request->get('user_email'))
                ->title('Search')
        ];
    }
```

The on my TestListScreen.php's `query()` function
```php
return ['results' => TestResult::filtersApply([UserEmailFilter::class])->filters()->paginate()];
```

Then, on my column: 
```php
TD::make('user.email', 'Email')
->filter(UserEmailFilter::class)
->render(function (TestResult $result) {
                    return $result->user->email;
                })
```

However, I'm only getting the basic text filter show up and not the one from the `display()` function on my custom one. Of course, text filtering is not working at all since the `run()` function is never executing.

I'm finding the official docs quite confusing. For instance, it mentions the `filtersApply()` function but the code sample does not use it. What am I missing? Is it actually possible to make this work for that specific column or does it have to be through a Selection? Any help appreciated. 🙏

**Comment by tabuna** [comment]
```php
TD::make('user.email', 'Email')
->filter(UserEmailFilter::class). // to https://example.com?filter[user.email]=any
```

The filter method in TD adds a prefix, so it does not fit your parameters specified to the filter class

**Reply by caposaric** [reply]
Hey @tabuna, thanks for responding. That was actually a typo from my side, but the parameter name on the code actually matches the name of the field and it is still not working. As I said, it is not even rendering the Input on the filter's `display()` function.

Is there somewhere else I should be "registering" the filter to make this work? It doesn't error out, it just doesn't show at all and returns the default text filter instead.

**Reply by tabuna** [reply]
What version of the package are you using?

**Reply by caposaric** [reply]
I'm working on Orchid v13.10
